DECISION-RECORD.md

```diff
@@ -0,0 +1,292 @@
+- No need to escape special characters in the filter value when operator is taken as a literal filter term. 
+
+#### Cons
+
+- More work on server side to parse and group the filters. Custom URL parameter binder or parser to split the query string key into two components (field name, operator) may be needed
+- Special characters in variable names can be awkward. Custom binder to split the query string key into two components may be needed
+- Hard to manage custom combinational filters since there is an implicit AND among filters. How to manage OR filters?
+
+### RHS Colon
+
+Retrieves all the items where the price is greater than or equal to 10, but less than or equal to 100
+
+`GET /items??price=gte:10&price=lte:100`
+
+#### Pros
+
+- Easiest to parse on server side especially if duplicate filters are not supported. 
+  - No custom binders needed. 
+  - Many API frameworks already handle URL parameter arrays. 
+  - Multiple price filters will be under the same variable ‘price’ which may be a Sequence or Map. 
+
+#### Cons
+
+Literal values need special handling.
+
+Retrieve all items where the user_id is greater than 100. 
+
+`GET /items?user_id=gt:100 `
+
+However, if `gt:100` was a valid id, we could not find the items where the `user_id` equals `gt:100`
+
+### Search query param
+
+Retrieves all the items containing the terms red chair and the price is greater than or equal to 10 and less than or equal to 100
+
+`GET /items?q=title:red chair AND price:[10 TO 100]`
+
+- Adding filters and ranges with the search parameter
+  - Lucene syntax or ElasticSearch Simple Query Strings would be easy to support directly - if needed.
+
+#### Pros
+
+- Most flexible queries for API users
+- Almost no parsing required on backend, can pass directly to search engine or database
+
+#### Cons
+
+- Harder for beginners (Lucene syntax)
+- Full-text search doesn’t make sense for all resources. E.g. Fuzziness and term boosting doesn’t make sense for time series metric data.
+- Requires URL percent encoding i.e. use of cURL or Postman becomes more complicated.
+
+---
+
+## Badges API
+
+**Decision: Implemented as a top level API endpoint during prototyping, removed from scope for feedback round**
+
+History links:
+
+- https://phabricator.wikimedia.org/T65486
+- https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/125886
+- https://global-search.toolforge.org/?q=availablebadges&namespaces=2%2C4%2C8&title=%28Gadgets-definition%7C.*%5C.%28js%7Ccss%7Cjson%29%29 (not used in mid 2020)
+- https://codesearch.wmcloud.org/search/?q=wbavailablebadges&i=fosho&files=&repos= (not used in mid 2020)
+
+---
+
+## Updating resources
+
+**Decision: Use PATCH and json-patch **
+
+### Patch
+
+When deciding on the update strategy of resources in our API we took into consideration two different proposals for sending updates: Payload comparison
```

Comment:
Might need a few more words here about why we chose to use PATCH (wikidata specifically works with big objects, etc...)